### PR TITLE
CustomSelectControl: Add flag for larger default size

### PR DESCRIPTION
**Pull request by mirka**
Part of #39397

## What?

Adds an internal `__next36pxDefaultSize` feature flag prop to introduce the larger default size for `CustomSelectControl`.

## Why?

In preparation for standardizing the default sizes to 36px height.

## How?

Following the procedure outlined in #39397, conditional styles have been added to enlarge the height and paddings. The font size is also enlarged to be consistent with `SelectControl`.

The inconsistent chevron (#39400) will be addressed separately.

@jasmussen How do the paddings feel?

## Testing Instructions

1. `npm run storybook:dev` and see the stories for `CustomSelectControl`.
2. From the Controls tab, toggle the `__next36pxDefaultSize` prop.

### `__next36pxDefaultSize` = false
Current default size

### `__next36pxDefaultSize` = true

_Updated with 16px padding:_

Larger default size

### CODE REVIEW

**Comment by jasmussen** [comment]
Nice, thank you for the clear instructions. This is what I see in Storybook:
select

Looks good, looks more coherent. 

Just for anyone else reviewing, we've had conversations about standardizing the heights across all inputs to be 36px, potentially with an option for 40px as well in the future. As we continue to refine those inputs and sizes, just unifying it all under a single 36px value is a good start, and will make future changes simpler.

In that light, do we need to include `36px` in the flag at all? I.e. could it simpluy be called `__nextDefaultSize`? While the current plan is to standardize all on 36px and have 40px as an option, we could go with `__nextDefaultSize` and `__nextLargeSize` instead, and it'd be a bit more agnostic to the values. 

On the padding, here's a screenshot of the resting state:

resting state of dropdown

Here's one of the target designs:
target design

Here's a direct comparison of the metrics:
metrics

A few small tweaks needed here. The left and right padding should be 16px (`$grid-unit-20` in the SCSS). For the chevron icon, we might need to pull it to the right a bit to massage the pixels, but it becomes visible when you compare with a input with the custom unit selector:
custom units

dropdown

There's a new chevronDownSmall icon, which we can use to better match the design, shown on the right here:

chevronicons

The color of the border (both for the dropdown and all input fields) should be #949494, or `$gray-600` in the SCSS system. That matches 3:1 UI contrast ratio.

You note an alignment issue with the chevron needing a separate fix, sounds good, I don't think we need to do all of the above feedback in a single PR, but I figured I'd write it down regardless! Hope that's useful, thanks again!

File: packages/components/src/custom-select-control/index.js [hunk 1]
```diff
@@ -141,8 +143,11 @@ export default function CustomSelectControl( {
 					// This is needed because some speech recognition software don't support `aria-labelledby`.
 					'aria-label': label,
 					'aria-labelledby': undefined,
-					className: 'components-custom-select-control__button',
-					isSmall: true,
+					className: classnames(
+						'components-custom-select-control__button',
+						{ 'is-next-36px-default-size': __next36pxDefaultSize }
+					),
+					isSmall: ! __next36pxDefaultSize,
```
Comment (ciampo) [inline packages/components/src/custom-select-control/index.js:150]: Love it when we rely on composability instead of style overrides

**Review by ciampo — COMMENTED** [review]
Code changes LGTM, and thank you for refactoring the Story to use Controls!

I'll let you and @jasmussen discuss and refine the component's spacing!

> In that light, do we need to include 36px in the flag at all? I.e. could it simply be called `__nextDefaultSize`? While the current plan is to standardize all on 36px and have 40px as an option, we could go with `__nextDefaultSize` and `__nextLargeSize` instead, and it'd be a bit more agnostic to the values.

I don't have a strong opinion here, both options sound good to me,

**Comment by mirka** [comment]
Thanks for the reviews!

> In that light, do we need to include `36px` in the flag at all? I.e. could it simply be called `__nextDefaultSize`?

Good question. It isn't pretty, but the reason I prefer including a specific pixel value here is twofold:

1. For anyone who is not familiar with our high-level component plans, I think that specificity is actually useful. Even for me, I get confused with all the size differences and the verbose naming decreases my cognitive load.
2. I want to avoid creating a situation where this flag ends up adding even more chaos to the library. The main thing I'm hedging against is when these size migrations take much longer than expected to complete (e.g. I'm hit by a bus, or priorities change). I don't want a vague `__nextDefaultSize` flag lying around when the next design refresh comes along 😂

> The left and right padding should be 16px (`$grid-unit-20` in the SCSS).

Do you have any concerns about this diverging from the `Button` paddings (12px) at 36px height? For example if a `Select` and a `Button` were stacked vertically, the paddings would be slightly off.

> The color of the border (both for the dropdown and all input fields) should be #949494, or `$gray-600` in the SCSS system.

Good to know. I've split out a separate issue (#39425) so we can change all of them at once.

**Comment by jasmussen** [comment]
> Do you have any concerns about this diverging from the Button paddings (12px) at 36px height? For example if a Select and a Button were stacked vertically, the paddings would be slightly off.

Excellent eye for detail. Thank you.

It's my sense that we'd want to standardize on 16px padding left and right for all of these 36px items, including Buttons (barring any responsive changes we might make). That's perhaps the biggest benefit this effort has: unifying all these drifted values across a single grid.

I'd think it okay to not do those changes all at once: in general we should avoid stacking dropdows and buttons vertically, and the places we do need love regardless.

**Comment by mirka** [comment]
> It's my sense that we'd want to standardize on 16px padding left and right for all of these 36px items, including Buttons

Gotcha, I've split out the `Button` padding issue to #39431 👍 

As for this PR, I went ahead and changed it to the wider padding (https://github.com/WordPress/gutenberg/pull/39401/commits/126c3d999d184c526efc275d0ceec2aedbc2858b). I'll deal with the right side padding when I fix the chevron problem (#39400).

---

With that, I think all feedback has been accounted for and this PR is ready for final approval 🙏

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

- Horizontal padding has been adjusted to `16px` in this PR
- Follow-ups:
    - Chevron icon inconsistency tracked separately in #39400
    - Button padding inconsistency tracked separately in #39431
    - Border color inconsistencies tracked separately in #39425

**Pull request merged**